drivers/clock_control/clock_stm32g0.c

```diff
@@ -27,31 +26,64 @@
 #define pllr(v) z_pllr(v)
 
 /**
- * @brief Set up pll configuration
+ * @brief Return PLL source
  */
-int config_pll_sysclock(void)
+__unused
+static uint32_t get_pll_source(void)
 {
-	uint32_t pll_source, pll_m, pll_n, pll_r;
-
-	pll_n = STM32_PLL_N_MULTIPLIER;
-	pll_m = pll_div(STM32_PLL_M_DIVISOR);
-	pll_r = pllr(STM32_PLL_R_DIVISOR);
-
 	/* Configure PLL source */
 	if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
-		pll_source = LL_RCC_PLLSOURCE_HSI;
+		return LL_RCC_PLLSOURCE_HSI;
 	} else if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
-		pll_source = LL_RCC_PLLSOURCE_HSE;
-	} else {
+		return LL_RCC_PLLSOURCE_HSE;
+	}
+
+	return -ENOTSUP;
+}
+
+/**
+ * @brief Set up pll configuration
+ */
+__unused
+int config_pll_sysclock(void)
+{
+	uint32_t pll_source;
+
+	pll_source = get_pll_source();
+	if (pll_source < 0) {
```

Comment:
`get_pll_source()` returns `uint32_t`.
Therefore, `if (pll_source < 0)` is always `false`.